lit-action-claim-key/nodejs/README.md

```diff
@@ -0,0 +1,35 @@
+# Claim a Deterministically Derived PKP Using a User ID
+
+This code demonstrates how to use a custom user ID to deterministically generate a PKP.
+
+You can read more about claimable keys [here](https://developer.litprotocol.com/user-wallets/pkps/claimable-keys/intro), and how to claim keys using a Lit Action [here](https://developer.litprotocol.com/sdk/serverless-signing/key-claiming).
+
+## Running the Example
+
+### Install the Dependencies
+
+In this directory, `lit-action-claim-key/nodejs`, run `yarn` to install the project dependencies.
+
+### Setting Up the `.env` File
+
+Make a copy of the provided `.env.example` file and name it `.env`:
+
+```
+cp .env.example .env
+```
+
+Within the `.env` there are the ENVs:
+
+1. `ETHEREUM_PRIVATE_KEY` - **Required**
+   - Must have Lit test tokens on the Chronicle Yellowstone blockchain
+     - [Faucet for Chronicle Yellowstone](https://chronicle-yellowstone-faucet.getlit.dev/)
+   - Will be used to:
+     - Mint PKP
+
+### Running the Tests
+
+After the `.env` is configured, the `test` NPM script in the `package.json` will run a test that executes the example script. Run it using:
+
+```
+yarn test
+```
```

Comment:
Maybe we should add exactly what the code is doing here:

1. Using an imported Ethereum private key, connect the wallet to the a RPC endpoint `Chronicle Yellowstone`. This will be used to pay for Lit usage
2. Use the `LitNodeClient` to connect to the Lit network
3. Connect the `LitContracts` client to the Lit network (`datil-test` in this case)
4. **If not provided in the .env file**: Mint a [`capacityCreditsNFT`](https://developer.litprotocol.com/sdk/capacity-credits) and define the request limit and expiration date
5. Create a `capacityDelegationAuthSig`. Any network costs will be undertaken by the `dAppOwnerWallet`
6. Generate the session signatures via an [`AuthSig`](https://v6-api-doc-lit-js-sdk.vercel.app/interfaces/types_src.AuthSig.html) (Authentication Signature)
7. Create a `userId` by combining an arbitrary string, which could be a username, and the current timestamp. This is to ensure the `userId` is unique
8. Execute the [Lit Action](./litAction.ts). The Lit Action will derive a `keyId` from the `userId`
8. Use the `LitContracts.pubkeyRouterContract.read.getDerivedPubkey` method to derive a PKP public key from the `keyId`
9. Use the `litContractClient.pkpNftContract.write.claimAndMint` method to claim the derived PKP public key, and mint a new PKP that has the same public key
10. Wait for the transaction to finish, and return the PKP public key, token ID, and ETH address